File: Leetcode/Dynamic Programming/#70 Climbing Stairs.cpp
class Solution {
public:
    int dp[50]{};
    int climbStairs(int n) {
     if(n==0) return 1;
     if(n==1) return 1;
        if(dp[n]) return dp[n];
        dp[n]=climbStairs(n-1)+climbStairs(n-2);
        return dp[n];
        
    }
};

File: 알고리즘수업/DP/#70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        cache = [-1] * (n + 1)
        
        def topdown(i):
            if i <= 1:
                return 1
            if cache[i] is not -1:
                return cache[i]
            
            cache[i] = topdown(i-1) + topdown(i-2)
            return cache[i]
        
        return topdown(n)
        